src/nl/hannahsten/texifyidea/inspections/latex/codematurity/LatexPrimitiveStyleInspection.kt

```diff
@@ -59,40 +61,53 @@ class LatexPrimitiveStyleInspection : TexifyInspectionBase() {
         return descriptors
     }
 
-    private inner class InspectionFix : LocalQuickFix {
+    private inner class InspectionFix(val oldCommand: SmartPsiElementPointer<LatexCommands>) : LocalQuickFixAndIntentionActionOnPsiElement(oldCommand.element) {
 
         @Nls
         override fun getFamilyName(): String {
             return "Convert to LaTeX alternative"
         }
 
-        override fun applyFix(
-            project: Project,
-            descriptor: ProblemDescriptor
-        ) {
-            val element = descriptor.psiElement as? LatexCommands ?: return
+        override fun getText(): String {
+            return oldCommand.element
+                ?.let { "Convert ${it.name} to ${CommandMagic.stylePrimitiveReplacements[it.name]}" }
+                ?: familyName
+        }
 
-            // Find elements that go after the primitive.
-            val cmdIndex = CommandMagic.stylePrimitives.indexOf(element.name)
-            if (cmdIndex < 0) {
-                return
-            }
-            val content = element.parentOfType(LatexNoMathContent::class)
-            val next = content!!.nextSiblingIgnoreWhitespace()
-            val after = if (next == null) "" else next.text
-            val replacement =
-                String.format(CommandMagic.stylePrimitveReplacements[cmdIndex], after)
-            val document =
-                PsiDocumentManager.getInstance(project).getDocument(element.containingFile)
+        override fun invoke(project: Project, file: PsiFile, editor: Editor?, startElement: PsiElement, endElement: PsiElement) {
+            (startElement as? LatexCommands)?.let { command ->
+                command.setName(CommandMagic.stylePrimitiveReplacements[command.name] ?: return)
```

Comment:
I think this setName might be problematic. I don't see why you need it either, you set the name but you're not using that (since you're providing text to the psi helper).

Example file:
```
\documentclass{article}


\begin{document}
asdf {my \textbf bold text} not bold
\end{document}
```

Result after quickfix:
```
\documentclass{article}


\begin{document}
asdf {my \textbf bold text} not bold
\end{document}
```

```
java.lang.Throwable: file/doc text length different, File[LaTeX source file main.tex, Language: Latex, com.intellij.psi.SingleRootFileViewProvider{vFile=file:///home/thomas/GitRepos/random-tex/src/main.tex, vFileId=310, content=VirtualFileContent{size=94}, eventSystemEnabled=true}] file.length=93; doc.length=94
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:202)
```

```
Inconsistent FILE tree in com.intellij.psi.SingleRootFileViewProvider{vFile=file:///home/thomas/GitRepos/random-tex/src/main.tex, vFileId=310, content=VirtualFileContent{size=90}, eventSystemEnabled=true}; nodeLength=93; fileLength=94

java.lang.Throwable
	at com.intellij.idea.IdeaLogger.error(IdeaLogger.java:127)
```

main.document.txt
```
\documentclass{article}


\begin{document}
asdf {my \textbf bold text} not bold
\end{document}
```

Reply:
I've separated the cases `\bf{arg}`  and {\bf bold}` now. In the first case all we have to do is change the name of the argument, so that's what the `setName` is for.

The exception occurred because I forgot a `.firstChild` after creating the psi. Not sure why it wasn't crashing when I was testing before.